### ISSUE TITLE
4.2 basics: sneaky `as_ref` introduced with no explanation

### ISSUE THREAD

**Issue by mackwic** [Gankro/too-many-lists#26]
In file https://github.com/Gankro/too-many-lists/blob/master/text/third-basics.md

line 66:
```rust
    List { head: self.head.map(|node| node.next.clone()) }
```

line 78-79
```rust
src/third.rs:28         List { head: self.head.as_ref().map(|node| node.next.clone()) }
                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

line 90:
```rust
    List { head: self.head.as_ref().and_then(|node| node.next.clone()) }
```

Seems like we passed from `head.map` to `head.as_ref().map` with no explanation

**Comment by Gankro** [comment]
I think line 66 was a typo, and I always intended to have `as_ref()`. We're far enough along at that pointe in the book that constantly showing that I keep forgetting to call `as_ref` isn't super helpful.

**Comment by mackwic** [comment]
Maybe not keeping forgetting `as_ref`, just a side not making a reminder about the why and how it applies here. Not superfluous I think

**Comment by Gankro** [comment]
Yeah sure. Just reasoning that I never meant to leave it out in this case at all.
